decidim-core/app/controllers/concerns/decidim/needs_tos_accepted.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Decidim
+  # Shared behaviour for signed_in users that require the latest TOS accepted
+  module NeedsTosAccepted
+    def tos_accepted_by_user
+      return true unless current_user
+      return if current_user.tos_accepted?
+      return if permited_paths?
+
+      redirect_to_tos
+    end
+
+    private
+
+    def permited_paths?
+      permited_paths = [tos_path, decidim.delete_account_path, decidim.accept_tos_path]
+      case request.path
```

Comment:
`permitted_paths.include?(request.path)` instead of all this `case`

Reply:
yes, its better